guppylang/module.py

```diff
@@ -325,19 +326,19 @@ def compile(self) -> Package:
             return self._compiled_hugr
 
         self.check()
+        checked_defs = self._imported_checked_defs | self._checked_defs
```

Comment:
I removed `def _compile_defs` as the old code (removed here) was the only place calling it